src/Plug.vala

```diff
@@ -273,80 +273,74 @@ public class About.Plug : Switchboard.Plug {
         main_grid.add (button_grid);
         main_grid.show_all ();
     }
-}
 
-private void reset_all_keys (GLib.Settings settings) {
-    var keys = settings.list_keys ();
-    foreach (var key in keys) {
-        settings.reset (key);
-    }
-}
+     /**
+     * returns true to continue, false to cancel
+     */
+    private bool confirm_restore_action () {
+        var dialog = new RestoreDialog ();
+        dialog.show_all ();
+
+        var result = dialog.run ();
+        dialog.destroy ();
 
-private string[] get_pantheon_schemas () {
-    string[] schemas = {};
-    string[] pantheon_schemas = {};
-    string[] prefixes = { "org.pantheon.desktop", "org.gnome.desktop" };
+        return result == 1;
+    }
 
-    var sss = SettingsSchemaSource.get_default ();
+    private void settings_restore_clicked () {
+        var should_display = confirm_restore_action ();
 
-    sss.list_schemas (true, out schemas, null);
+        if (should_display) {
+            var all_schemas = get_pantheon_schemas ();
 
-    foreach (var schema in schemas) {
-        foreach (var prefix in prefixes) {
-            if (schema.has_prefix (prefix)) {
-                pantheon_schemas += schema;
+            foreach (var schema in all_schemas) {
+                reset_recursively (schema);
             }
         }
     }
-    return pantheon_schemas;
-}
-
-private void reset_recursively (string schema) {
-    var settings = new GLib.Settings (schema);
-    // change into delay mode
-	// so changes take place when apply () is called
-    settings.delay ();
-
-    reset_all_keys (settings);
-
-    var children = settings.list_children ();
-    foreach (var child in children) {
-        var child_settings = settings.get_child (child);
 
-        reset_all_keys (child_settings);
+    private static void reset_all_keys (GLib.Settings settings) {
+        var keys = settings.list_keys ();
```

Comment:
Is this variable necessary?  Only used once and does not increase clarity significantly imo.